src/PredictorMfe2dHeuristic.cpp

```diff
@@ -98,36 +98,38 @@ fillHybridE()
 {
 	// compute entries
 	// current minimal value
-	E_type curE = E_INF, curEtotal = E_INF, curCellEtotal = E_INF;
-	size_t i1,i2,w1,w2;
-	BestInteraction * curCell = NULL;
-	const BestInteraction * rightExt = NULL;
+	
 	// iterate (decreasingly) over all left interaction starts
-	for (i1=hybridE.size1(); i1-- > 0;) {
-	for (i2=hybridE.size2(); i2-- > 0;) {
+	#pragma omp parallel for
```

Comment:
Given the logic of the underlying recursion, you can only parallelize the "i2-loop". For a fixed i1, all i2 related computations are independent..